pair/src/lib.rs

```diff
@@ -0,0 +1,498 @@
+#![no_std]
+
+use gear_lib::fungible_token::{ft_core::*, state::*};
+use gear_lib_derive::{FTCore, FTMetaState, FTStateKeeper};
+use gstd::{cmp, exec, msg, prelude::*, ActorId};
+use num::integer::Roots;
+use pair_io::*;
+pub mod math;
+pub mod messages;
+
+const MINIMUM_LIQUIDITY: u128 = 1000;
+static ZERO_ID: ActorId = ActorId::zero();
+
+#[derive(Debug, Default, FTStateKeeper, FTCore, FTMetaState)]
+pub struct Pair {
+    #[FTStateField]
+    pub token: FTState,
+    pub factory: ActorId,
+    pub token0: ActorId,
+    pub token1: ActorId,
+    last_block_ts: u128,
+    pub balance0: u128,
+    pub balance1: u128,
+    reserve0: u128,
+    reserve1: u128,
+    pub price0_cl: u128,
+    pub price1_cl: u128,
+    pub k_last: u128,
+}
+
+static mut PAIR: Option<Pair> = None;
+
+impl Pair {
+    // INTERNAL METHODS
+
+    /// Mints the liquidity.
+    /// `to` - MUST be a non-zero address
+    /// Arguments:
+    /// * `to` - is the operation performer
```

Comment:
Doc comments for private methods are not valid.
```suggestion
    // Mints the liquidity.
    // `to` - MUST be a non-zero address
    // Arguments:
    // * `to` - is the operation performer
```